Re-implement using pip-tools

AdrianDAlessandro opened the pull request:
This PR reimplements the poetry template as a pip-tools based template to provide us the alternative.

AdrianDAlessandro committed:
Re-implement using pip-tools

AdrianDAlessandro committed:
3.9 -> 3.10 in Readme

alexdewar reviewed (approved):
LGTM!

Why did you update the config files for the check links stuff btw? Was it not working for you?

dalonsoa reviewed (approved):
Looks great!

AdrianDAlessandro committed:
Update ci.yml

AdrianDAlessandro commented:
> Why did you update the config files for the check links stuff btw? Was it not working for you?

It was working, but there was needless repetition but having one activated on push in `ci.yml` and a separate one for scheduled runs. I removed the push-activated one to reduce clutter in `ci.yml` and added the ability for it to run on push in the scheduled one.

AdrianDAlessandro committed:
Update ci.yml

AdrianDAlessandro committed:
Specify how to restrict package versions

cc-a commented on README.md:
```diff
@@ -1,45 +1,67 @@
-This is a minimal Python 3.9 application that uses [poetry](https://python-poetry.org) for packaging and dependency management. It also provides [pre-commit](https://pre-commit.com/) hooks (for [isort](https://pycqa.github.io/isort/), [Black](https://black.readthedocs.io/en/stable/), [Flake8](https://flake8.pycqa.org/en/latest/) and [mypy](https://mypy.readthedocs.io/en/stable/)) and automated te ...
+This is a minimal Python 3.10 application that uses [`pip-tools`] for packaging and dependency management. It also provides [`pre-commit`](https://pre-commit.com/) hooks (for [`isort`](https://pycqa.github.io/isort/), [`black`](https://black.readthedocs.io/en/stable/), [`flake8`](https://flake8.pycqa.org/en/latest/) and [`mypy`](https://mypy.readthedocs.io/en/stable/)) and automated tests using [ ...
+
+[`pip-tools`] is chosen as a lightweight dependency manager that adheres to the [latest standards](https://peps.python.org/pep-0621/ using `pyproject.toml`).
```
The link at the end doesn't look right.

AdrianDAlessandro replied:
Interesting that the combination of markdownlint and check-links action didn't pick up on that

cc-a commented on README.md:
````diff
@@ -1,45 +1,67 @@
    ```bash
-   poetry install
+   python -m venv .venv
+   source .venv/bin/activate # on Windows: `.venv\Scripts\Activate.ps1`
````
```suggestion
   source .venv/bin/activate # with Powershell on Windows: `.venv\Scripts\Activate.ps1`

```

cc-a reviewed (approved):
Awesome. Thanks for this @AdrianDAlessandro. Couple of very tiny comments but otherwise good to go.

AdrianDAlessandro committed:
Update README.md

Co-authored-by: Christopher Cave-Ayland <c.cave-ayland@imperial.ac.uk>

AdrianDAlessandro committed:
Fix link
